src/main/java/com/amazonaws/ant/AWSAntTask.java

```diff
@@ -58,28 +65,55 @@ public void setAWSAccessKeyId(String awsAccessKeyId) {
     public void setAWSSecretKey(String awsSecretKey) {
         this.awsSecretKey = awsSecretKey;
     }
-
+    
+    /**
+     * Sets AWS Region.
+     * 
+     * @param awsRegion
+     *            The AWS Region to set.
+     */
+    public void setAWSRegion(String awsRegion) {
+        this.awsRegion = awsRegion;
+    }
     
     @SuppressWarnings("unchecked")
     public <T extends AmazonWebServiceClient> T getOrCreateClient(
             Class<T> clientClass) {
         if(getProject().getReference(CLIENT_CACHE_REFERENCE) == null) {
-            Map<Class<?>, Object> cache = 
-                    new HashMap<Class<?>, Object>();
+            Map<List<String>, Object> cache = new HashMap<List<String>, Object>();
             getProject().addReference(CLIENT_CACHE_REFERENCE, cache);
         }
-        Map<Class<?>, Object> cache =
-                getProject().getReference(CLIENT_CACHE_REFERENCE);
-        T client = (T) cache.get(clientClass);
+        Map<List<String>, Object> cache = getProject().getReference(CLIENT_CACHE_REFERENCE);
+        
+        List<String> key = Arrays.asList(clientClass.getName(), this.awsRegion);
+        
+        T client = (T) cache.get(key);
         if(client == null) {
             T newClient = createClient(clientClass);
-            cache.put(clientClass, newClient);
+            Region region = getRegion();
+            if(region != null) {
+                newClient.setRegion(region);
+            }
+            cache.put(key, newClient);
             return newClient;
         } else {
             return client;
         }
     }
     
+    private Region getRegion() {
+        if(StringUtils.isNullOrEmpty(this.awsRegion)) {
+            return null;
+        }
+        Regions regions = Regions.DEFAULT_REGION;
+        try {
+            regions = Regions.fromName(this.awsRegion);
+        } catch(IllegalArgumentException ex) {
+            throw new BuildException("Error in parameter configuration: The specified awsRegion [" + this.awsRegion + "] is not valid\n");
+        }
+        return Region.getRegion(regions);
```

Comment:
Can you call [RegionUtils.getRegion(String)](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/regions/RegionUtils.html#getRegion(java.lang.String)) directly here instead of bouncing through the Regions enum? That'll make it easier for people to use new regions without having to wait until this project is updated to use a version of the SDK whose Regions enum includes the new value.